mmcls/models/heads/arcface_head.py

```diff
@@ -0,0 +1,217 @@
+
+        if self.easy_margin:
+            # when cosine>0, choose phi
+            # when cosine<=0, choose cosine
+            phi = torch.where(cosine > 0, phi, cosine)
+        else:
+            # when cos>th, choose phi
+            # when cos<=th, choose cosine-mm
+            phi = torch.where(cosine > self.th, phi, cosine - self.mm)
+
+        one_hot = torch.zeros(cosine.size(), device=pre_logits.device)
+        one_hot.scatter_(1, target.view(-1, 1).long(), 1)
+        if self.ls_eps > 0:
+            one_hot = (1 -
+                       self.ls_eps) * one_hot + self.ls_eps / self.num_classes
+
+        output = (one_hot * phi) + ((1.0 - one_hot) * cosine)
+
+        output *= self.s
+
+        return output
+
+    def loss(self, feats: Tuple[torch.Tensor],
+             data_samples: List[ClsDataSample], **kwargs) -> dict:
+        """Calculate losses from the classification score.
+
+        Args:
+            feats (tuple[Tensor]): The features extracted from the backbone.
+                Multiple stage inputs are acceptable but only the last stage
+                will be used to classify. The shape of every item should be
+                ``(num_samples, num_classes)``.
+            data_samples (List[ClsDataSample]): The annotation data of
+                every samples.
+            **kwargs: Other keyword arguments to forward the loss module.
+
+        Returns:
+            dict[str, Tensor]: a dictionary of loss components
+        """
+
+        if 'score' in data_samples[0].gt_label:
+            # Batch augmentation may convert labels to one-hot format scores.
+            target = torch.stack([i.gt_label.score for i in data_samples])
+        else:
+            target = torch.hstack([i.gt_label.label for i in data_samples])
+
+        # The part can be traced by torch.fx
+        cls_score = self(feats, target)
+
+        # compute loss
+        losses = dict()
+        loss = self.loss_module(
+            cls_score, target, avg_factor=cls_score.size(0), **kwargs)
+        losses['loss'] = loss
+
+        return losses
+
+    def predict(self,
+                feats: Tuple[torch.Tensor],
+                data_samples: List[ClsDataSample] = None) -> torch.Tensor:
+        """Inference without augmentation.
+
+        Args:
+            feats (tuple[Tensor]): The features extracted from the backbone.
+                Multiple stage inputs are acceptable but only the last stage
+                will be used to classify. The shape of every item should be
+                ``(num_samples, num_classes)``.
+            data_samples (List[ClsDataSample], optional): The annotation
+                data of every samples. If not None, set ``pred_label`` of
+                the input data samples. Defaults to None.
+
+        Returns:
+            torch.Tensor: features of samples.
+        """
+
+        pre_logits = self.pre_logits(feats)
+        if self.used == 'after':
+            pre_logits = self.norm_layer(pre_logits)
```

Comment:
output from norm layer is scaled. 

```
pre_logits = self.s * self.norm_layer(pre_logits)
```

Reply:
Thanks for your reminding, the modification has been made.